File: iCinema.Application/DTOs/Reservations/TicketDto.cs
namespace iCinema.Application.DTOs.Reservations;

public class TicketDto
{
    public Guid TicketId { get; set; }
    public string? QRCode { get; set; }
    public string? TicketStatus { get; set; }
    public string? TicketType { get; set; }
    public int RowNumber { get; set; }
    public int SeatNumber { get; set; }
}

File: iCinema.UI/icinema_mobile_client/lib/features/reservations/data/models/ticket_dto.dart
class TicketDto {
  final String ticketId;
  final String? qrCode;
  final String ticketStatus; // Active | Used | Canceled
  final String? ticketType;
  final int rowNumber;
  final int seatNumber;

  const TicketDto({
    required this.ticketId,
    required this.qrCode,
    required this.ticketStatus,
    required this.ticketType,
    required this.rowNumber,
    required this.seatNumber,
  });

  factory TicketDto.fromJson(Map<String, dynamic> json) => TicketDto(
        ticketId: json['ticketId']?.toString() ?? '',
        qrCode: json['qrCode'] as String?,
        ticketStatus: json['ticketStatus']?.toString() ?? 'Active',
        ticketType: json['ticketType'] as String?,
        rowNumber: (json['rowNumber'] as num?)?.toInt() ?? 0,
        seatNumber: (json['seatNumber'] as num?)?.toInt() ?? 0,
      );
}
